### PR TITLE
Fix #353 - Add settings option for password length. 

### LINKED ISSUE #353: Random password length
https://github.com/rainlab/user-plugin/blob/efa31448a9fadb0fb72fefaccd4c5b179998d451/models/User.php#L421

Why is the generated Random Password only 6 chars long?

### PR DESCRIPTION
**Pull request by obuchmann**
- Add settings option for password length. 
- Use password length for password validation and password generation.

### CODE REVIEW

File: models/Settings.php [hunk 1]
```diff
@@ -16,13 +16,18 @@ class Settings extends Model
     public $settingsCode = 'user_settings';
     public $settingsFields = 'fields.yaml';
 
+
+    const PASSWORD_LENGTH_KEY = 'password_length';
```
Comment (LukeTowers) [inline models/Settings.php]: Why is this required?

Comment (kiselli) [reply models/Settings.php]: It prevents developers from using String keys and helps refactoring the code later. But we can cut it off.

**Comment by bennothommo** [comment]
@obuchmann I have just done some minor code cleanup and renamed `password_length` to `min_password_length` where applicable to make it clear this is a minimum password length. I'll be happy to include this in release 1.5.0 of the User plugin after some testing. I have tested it myself and it works well.

File: models/Settings.php [hunk 2]
```diff
@@ -16,13 +14,17 @@ class Settings extends Model
...
 
+
     const ACTIVATE_AUTO = 'auto';
     const ACTIVATE_USER = 'user';
     const ACTIVATE_ADMIN = 'admin';
 
     const LOGIN_EMAIL = 'email';
     const LOGIN_USERNAME = 'username';
 
+    const MIN_PASSWORD_LENGTH = 'min_password_length';
```
Comment (LukeTowers) [inline models/Settings.php]: why is this a class constant? It seems unnecessary.

Comment (bennothommo) [reply models/Settings.php]: No idea, it was in the original PR. I've removed it.

**Comment by tobias-kuendig** [comment]
I just did a test run of these changes. Everything is working as expected :+1: 

Great work @obuchmann and @bennothommo 	!

File: models/User.php [hunk 3]
```diff
@@ -225,6 +234,14 @@ public function beforeValidate()
         ) {
             $this->username = $this->email;
         }
+
+
+        /*
+         * Apply Password Length Settings
+         */
+        $minPasswordLength = self::getMinPasswordLength();
```
Comment (LukeTowers) [inline models/User.php]: Can you change `self` to `static` so that people can override the `getMinPasswordLength()` method if they want to?

File: models/User.php [hunk 4]
```diff
@@ -418,6 +435,6 @@ protected function sendInvitation()
      */
     protected function generatePassword()
     {
-        $this->password = $this->password_confirmation = Str::random(6);
+        $this->password = $this->password_confirmation = Str::random(self::getMinPasswordLength());
```
Comment (LukeTowers) [inline models/User.php]: self to static please

**Comment by bennothommo** [comment]
Done @LukeTowers. :)
